src/components/provider/selectors.js

```diff
@@ -131,6 +139,28 @@ export const usePayload = () => {
   return chart.getPayload()
 }
 
+export const useChartError = () => {
+  const [error, setError] = useState(false)
+  const chart = useChart()
+  const forceUpdate = useForceUpdate()
+
+  useImmediateListener(
+    () =>
+      chart
+        .on("successFetch", () => {
+          setError(false)
+          forceUpdate()
+        })
+        .on("failFetch", () => {
+          setError(true)
+          forceUpdate()
+        }),
+    [chart]
+  )
```

Comment:
Since the only difference between fail and success is the error value, maybe having a `handleFetchError` function could be of help!

```suggestion
  const handleFetchError = hasError => {
    setError(hasError)
    forceUpdate()
  }
  
  useImmediateListener(
    () =>
      chart
        .on("successFetch", () => handleFetchError(false))
        .on("failFetch", () => handleFetchError(true)),
    [chart]
  )
```

Reply:
Agree with that. I would only change the name of `handleFetchError` to `handleFetch`

Reply:
That was my initial thinking too about the naming @AlexNti , but if you think of it... `handleFetch = hasError => etc.` might make sense of what it's actual work does... but `handleFetch(false/true)` doesn't really!

Reply:
I agree with @burbuli8ra

Reply:
For this reason we can add an object to params of `handleFetch` `handleFetch({hasError})` .